pubnub-core/pubnub-core-api/src/commonMain/kotlin/com/pubnub/api/models/consumer/objects/channel/PNChannelMetadata.kt

```diff
@@ -1,12 +1,29 @@
 package com.pubnub.api.models.consumer.objects.channel
 
+import com.pubnub.api.utils.PatchValue
+
 data class PNChannelMetadata(
     val id: String,
-    val name: String?,
-    val description: String?,
-    val custom: Map<String, Any?>?,
-    val updated: String?,
-    val eTag: String?,
-    val type: String?,
-    val status: String?,
-)
+    val name: PatchValue<String?>?,
+    val description: PatchValue<String?>?,
+    val custom: PatchValue<Map<String, Any?>?>?,
+    val updated: PatchValue<String>?,
+    val eTag: PatchValue<String>?,
+    val type: PatchValue<String?>?,
+    val status: PatchValue<String?>?,
+) {
+    /**
+     * Merge information from this `PNChannelMetadata` with new data from `update`, returning a new `PNChannelMetadata` instance.
+     */
+    operator fun plus(update: PNChannelMetadata): PNChannelMetadata {
```

Comment:
For me name "plus" is misleading. What do you think about "applyUpdate" or "combineWith" ?
and then to retain plus operator code could be modified like this:

```
    operator fun plus(update: PNChannelMetadata): PNChannelMetadata{
       return applyUpdate(update)
    }

    fun applyUpdate(update: PNChannelMetadata): PNChannelMetadata {
        return copy(
            name = update.name ?: name,
            description = update.description ?: description,
            custom = update.custom ?: custom,
            updated = update.updated ?: updated,
            eTag = update.eTag ?: eTag,
            type = update.type ?: type,
            status = update.status ?: status,
        )
    }
```

Reply:
regarding `applyUpdate` - it suggests that the object will change which is not the case, and regarding `combineWith` IMHO it literally means `plus`, so I don't really like any of the options, nor do I think that we need to have two functions doing the same thing in Kotlin, especially since the developer is supposed to do `a + b` and not use `plus` directly. However, the question may come up in the future when we expose this to Java, so I'm open to other suggestions for names, but so far I think it's not an improvement over `plus` (which is the required name for the operator override in Kotlin)

Reply:
For me this is not really "plus" operation but rather updateIfApplicable or just update.
oldDescritpion = "Old Desc"
newDescription = "New Desc" 

oldDescritpion + newDescription = "Old Desc New Desc"
oldDescritpion.updateIfApplicable(newDescription) = "New Desc"

Reply:
the plus operation can be many things depending on the domain: 
for a vector it can be what you described: (a,b,c) + (x,y,z) = (x+a, y+b, z+c)
for two lists it's [a] + [b, c] = [a, b, c] , (and not [ab, c])
for two maps it's {"a": "abc"} + {"a": "xyz"} = {"a": "xyz"} (right one wins)

for a change set (a data class containing what I called PatchValues) like we have here it's most similar two adding two maps, which can still be described as a plus operation

Reply:
what I'm saying is that the `plus` operator in Kotlin isn't always used for arithmetic plus or concatenation

Reply:
I see. Ok.